### ISSUE TITLE
What needs to be published

### ISSUE THREAD

**Issue by JohnHadish** [SystemsGenetics/GEMmaker#10]
Our publishDirs are out of control. I do not feel that we need to have anything in them besides the final fpkm and the fastqc files. Everything else is an intermediary file for these, and I don't think they are necessary. The workflow can be restarted even if we do not publish all of these intermediary files, as they are present in the work directory. The only disadvantage of this would be if these files are needed for an independent project from the workflow they are being run in.

**Comment by spficklin** [comment]
It would be nice to keep the BAM files around, but those are going to have all of the FASTQ data embedded so will still have 100TB of storage needs for the ATH experiment to keep that around.  We can use BAM files for visualization on the gene-networks.org website if we keep them.

thoughts @feltus?

**Comment by feltus** [comment]
GEMaker rulz.  
Can we tier data (irods policy)? -FASTQ-trim lives until BAMs are made.-BAMs expire in a year if nobody touches them.-GEMs and metadata never die.
Too simplistic?
Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Stephen Ficklin <notifications@github.com> Date: 6/1/18  7:06 PM  (GMT-05:00) To: SystemsGenetics/sra2gev <sra2gev@noreply.github.com> Cc: Alex Feltus <alex.feltus@gmail.com>, Mention <mention@noreply.github.com> Subject: Re: [SystemsGenetics/sra2gev] What needs to be published (#10) 
It would be nice to keep the BAM files around, but those are going to have all of the FASTQ data embedded so will still have 100TB of storage needs for the ATH experiment to keep that around.  We can use BAM files for visualization on the gene-networks.org website if we keep them.
thoughts @feltus?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.
{"@context":"http://schema.org","@type":"EmailMessage","potentialAction":{"@type":"ViewAction","target":"https://github.com/SystemsGenetics/sra2gev/issues/10#issuecomment-394031782","url":"https://github.com/SystemsGenetics/sra2gev/issues/10#issuecomment-394031782","name":"View Issue"},"description":"View this Issue on GitHub","publisher":{"@type":"Organization","name":"GitHub","url":"https://github.com"}}
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/SystemsGenetics/sra2gev","title":"SystemsGenetics/sra2gev","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/SystemsGenetics/sra2gev"}},"updates":{"snippets":[{"icon":"PERSON","message":"@spficklin in #10: It would be nice to keep the BAM files around, but those are going to have all of the FASTQ data embedded so will still have 100TB of storage needs for the ATH experiment to keep that around.  We can use BAM files for visualization on the gene-networks.org website if we keep them.\r\n\r\nthoughts @feltus?\r\n"}],"action":{"name":"View Issue","url":"https://github.com/SystemsGenetics/sra2gev/issues/10#issuecomment-394031782"}}}
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "37567f93-e2a7-4e2a-ad37-a9160fc62647",
"title": "Re: [SystemsGenetics/sra2gev] What needs to be published (#10)",
"sections": [
{
"text": "",
"activityTitle": "**Stephen Ficklin**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@spficklin",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"SystemsGenetics/sra2gev\",\n\"issueId\": 10,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close issue",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueClose\",\n\"repositoryFullName\": \"SystemsGenetics/sra2gev\",\n\"issueId\": 10\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/SystemsGenetics/sra2gev/issues/10#issuecomment-394031782"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 341672625\n}"
}
],
"themeColor": "26292E"
}

**Comment by spficklin** [comment]
Here's my suggestion for data management:

1)  After Trimmed FastQ files are generated and FASTQC reports are created delete the original Fastq files if they were downloaded from NCBI (do not delete original local files... copies can be deleted).
2)  After BAM files are generated delete the trimmed FastQ files. 
3)  After FPKM and TPM files are generated delete BAM files. If we do decide to create an account with AWS for their Public Data Repository then we can publish BAMs there.
4)  Publish FPKM, TPM, FastQC, GEMmaker Nextflow trace files, MultiQC reports for the entire run to IRODs.

**Comment by bentsherman** [comment]
So now that I've been generating data for the Ath experiment, it's becoming pretty important that we address this issue. I think we could integrate iRODS pretty easily here, either in the workflow itself or as a post-processing script. We just need to be able to build the icommands from source on Palmetto / Kamiak.

Here's my recommended strategy:

- Don't publish intermediate files. Like John said, these files are safe and sound in the work directory if we need to inspect them or resume a workflow multiple times. Nextflow only uses the work directory to recover cached results.
- Publish the final output files and then push them to iRODS via workflow / external script. That way the only interaction with iRODS is a bulk transfer once the workflow is complete and the results have been inspected.

I can implement this strategy once we have a working build of the icommands. As a side note, it looks like the fastq files from the Atk experiment compose pretty much all of the storage space. I imagine this will change once we fix the trimmed fastq and BAM / SAM files to not be empty.

**Comment by spficklin** [comment]
Hi @bentsherman , @cwytko has been working on this issue of publishing to iRODs we should talk.

**Comment by spficklin** [comment]
As per our 11/05/18 meeting, this should be fixed prior to @bentsherman PR #33 being merged.

We had discussed including these code fixes as part of PR #33 but I think it may be best to add the fix to this as a separate PR.

**Comment by bentsherman** [comment]
@JohnHadish I can tackle this if you want since this is now blocking my irods scripts from being merged.
